pkg/constraints/cel.go

```diff
@@ -0,0 +1,146 @@
+package constraints
+
+import (
+	"fmt"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/interpreter/functions"
+	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+
+	"github.com/blang/semver/v4"
+)
+
+// PropertiesKey is the key for bundle properties map (input data for CEL evaluation)
+const PropertiesKey = "properties"
+
+// Constraint is a struct representing the new generic constraint type
+type Constraint struct {
```

Comment:
Is the intent for `olm.constraint`'s definition to live in the api repo? I was under the impression that OLM's resolver would define this schema.

Reply:
I think it should be here given registry is gonna use it as well. So this is a shared code. I think Joe wants it here as well. I don't have a strong opinion about this. I can move it back to OLM and registry will have a copy of these 2 structs. It is not a big deal given the code is small.

Reply:
Right, I see it will be used [here](https://github.com/operator-framework/operator-registry/pull/879/files#diff-c3b6b7bfc976a561eb2abaf01a8f2de39d5c2ff616bc134daf695ae5d4eccd74R353) at least. SGTM.